app/src/actioncommands.cpp

```diff
@@ -713,11 +713,15 @@ void ActionCommands::moveFrameForward()
     Layer* layer = mEditor->layers()->currentLayer();
     if (layer)
     {
-        mEditor->notifyFrameMoved();
+        auto selectedFrames = layer->getSelectedFrames();
+        layer->deselectAll();
         if (layer->moveKeyFrame(mEditor->currentFrame(), 1))
         {
             mEditor->scrubForward();
-            mEditor->notifyFrameMoved();
+        }
+
+        for (int pos : selectedFrames) {
+            layer->setFrameSelected(pos, true);
```

Comment:
That would alter the functionality of the moveKeyframe method and may cause unwanted behavior, so I don't think that's a good idea. You may not want to always make a deselection and reselect afterwards.

> Additionally, layer stores the ordering of frame selections, but this is not the ordering returned by getSelectedFrames, so it will be lost when this operation is performed. If you move this to layer, you should be able to fix this and perhaps even simply things by manipulating the member variables directly.

Is this important here though? given that we're not dynamically modifying the list and or moving frames at this point, the order has no meaning?

Reply:
> Is this important here though? given that we're not dynamically modifying the list and or moving frames at this point, the order has no meaning?

I'm not sure what you mean by this. What I'm referring to is the order in which the frames are selected, which is stored in `Layer::mSelectedFrames_byLast`. Here's a gif showing how reordering with this deselection/reselection affects the behavior of further frame selections:
![Frame Selection Ordering](https://user-images.githubusercontent.com/3461051/102942105-ae3a9480-4471-11eb-974c-f103d981b9a0.gif)
In this I am clicking frame 5, then ctrl clicking frame 1, then shift clicking frame 3. The second time use the frame forward before the last step. As you can see, it then selects from the first frame selected (frame 5), not the last frame selected (frame 1).

Reply:
Ah you're right, I used the wrong selection variable. I've simplified the logic now and made it less ugly.